client/components/data/document-head/index.jsx

```diff
@@ -16,13 +16,26 @@ import {
 	addDocumentHeadMeta as addMeta,
 	setDocumentHeadUnreadCount as setUnreadCount
 } from 'state/document-head/actions';
-import { getSelectedSiteId } from 'state/ui/selectors';
 
 class DocumentHead extends Component {
 	componentWillMount() {
-		this.props.setTitle( this.props.title );
-		this.props.setDescription( this.props.description );
-		this.props.setUnreadCount( this.props.unreadCount );
+		const {
+			title,
+			description,
+			unreadCount
+		} = this.props;
+
+		if ( title ) {
+			this.props.setTitle( title );
+		}
+
+		if ( description ) {
+			this.props.setDescription( description );
+		}
+
+		if ( unreadCount ) {
+			this.props.setUnreadCount( String( unreadCount ) );
```

Comment:
Similar to https://github.com/Automattic/wp-calypso/pull/7224/files#r73967842, should we allow setting zero unread count? This won't be reached if passed `0` because of the condition.